src/Uno.AzureDevOps/Uno.AzureDevOps.Wasm/wwwroot/logincompleted.html

```diff
@@ -1,18 +1,25 @@
-﻿<!DOCTYPE html>
-<html>
-<head>
-    <meta charset="utf-8" />
-    <title></title>
-</head>
-<body>
-    The login has been completed!
-    <script type="text/javascript">
-        (function() {
-            const opener = window.opener;
-            const message = `nav|${window.location.href}`; // special message containing requested url
-            const origin = opener.location.origin; // opener should have same origin
-            opener.postMessage(message, origin);
-        })();
-    </script>
-</body>
-</html>
+﻿<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8" />
+    <title></title>
+</head>
+<body>
+    The login has been completed!
+    <script type="text/javascript">
+        (function() {
+            const opener = window.opener;
+            const message = `nav|${window.location.href}`; // special message containing requested url
+
+            if (opener && opener.location && opener.location.origin) {
+                const origin = opener.location.origin; // opener should have same origin
+
+                opener.postMessage(message, origin);
+            } else {
+                window.localStorage.setItem("__UadoToken", message);
+                window.close();
+            }


```

Comment:
`^M`  ???

Reply:
What's that ^M for ? Is it a typo or intended ?

Reply:
It's a CR without the coresponding LF.  That's an innofensive typo.